server/src/main/java/org/elasticsearch/search/internal/SearchContext.java

```diff
@@ -374,64 +759,672 @@ public void clearReleasables(Lifetime lifetime) {
+            this.timeout = timeout;
+            return this;
+        }
+
+        public Builder setTerminateAfter(int terminateAfter) {
+            this.terminateAfter = terminateAfter;
+            return this;
+        }
+
+        public Builder setExplain(boolean explain) {
+            this.explain = explain;
+            return this;
+        }
+
+        public Builder setVersion(boolean version) {
+            this.version = version;
+            return this;
+        }
+
+        public Builder setSeqAndPrimaryTerm(boolean seqAndPrimaryTerm) {
+            this.seqAndPrimaryTerm = seqAndPrimaryTerm;
+            return this;
+        }
+
+        public Builder setCollapse(CollapseContext collapse) {
+            this.collapse = collapse;
+            return this;
+        }
+
+        public Builder setScroll(ScrollContext scrollContext) {
+            this.scrollContext = scrollContext;
+            return this;
+        }
+
+        public Builder setStoredFields(StoredFieldsContext storedFields) {
+            this.storedFields = storedFields;
+            return this;
+        }
+
+        public Builder setScriptFields(ScriptFieldsContext scriptFields) {
+            this.scriptFields = scriptFields;
+            return this;
+        }
+
+        public Builder setFetchSource(FetchSourceContext fetchSourceContext) {
+            this.fetchSourceContext = fetchSourceContext;
+            return this;
+        }
+
+        public Builder setDocValueFields(DocValueFieldsContext docValueFieldsContext) {
+            this.docValueFieldsContext = docValueFieldsContext;
+            return this;
+        }
+
+        public Builder setInnerHits(Map<String, InnerHitContextBuilder> innerHits) {
+            this.innerHits = innerHits;
+            return this;
+        }
+
+        public Builder setGroupStats(List<String> groupStats) {
+            this.groupStats = groupStats;
+            return this;
+        }
+
+        public Builder setHighlight(SearchContextHighlight highlight) {
+            this.highlight = highlight;
+            return this;
+        }
+
+        public Builder setSuggest(SuggestionSearchContext suggest) {
+            this.suggest = suggest;
+            return this;
+        }
+
+        public Builder setRescorers(List<RescoreContext> rescorers) {
+            this.rescorers = rescorers;
+            return this;
+        }
+
+        public Builder setSearchExt(Map<String, SearchExtBuilder> builders) {
+            this.searchExts = builders;
+            return this;
+        }
+
+        public Builder setProfile(boolean profile) {
+            this.profile = profile;
+            return this;
+        }
+
         /**
-         * This life time is for objects that only live during collection time.
+         * Builds the highlight context from the provided builder.
          */
-        COLLECTION,
+        public Builder buildHighlight(HighlightBuilder builder) throws SearchException {
```

Comment:
I see why these methods return the Builder itself, but maybe this should be done only for the set methods? In some cases we also end up never using the return type.